java/arcs/sdk/android/storage/ServiceStore.kt

```diff
@@ -142,14 +150,19 @@ class ServiceStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
     override fun off(callbackToken: Int) {
         val service = checkNotNull(storageService)
         runBlocking {
-            channel.send { service.unregisterCallback(callbackToken) }
+            send { service.unregisterCallback(callbackToken) }
         }
+        initChannel()
```

Comment:
Will the very first line of initChannel: `channel?.let { it.cancel() }` happen to cancel the `send {}` after all the channel receive() the callback and executes it on a different coroutine scope.

Reply:
Good point. channel.cancel() actually just enqueues a channel.close() at the end of the queue. So we want to ensure service.unregisterCallback() runs. However to verify this, I'll write a quick test case on Kotlin playground.  
It could be that cancelling the scope used to collect the flow by stop the queue from being processed, in which case, we'd have to launch the flow in a scope that isn't cleaned up until the channel is closed.